src/cdomains/valueDomain.ml

```diff
@@ -142,7 +142,7 @@ struct
 
   let rec init_value (t: typ): t = (* top_value is not used here because structs, blob etc will not contain the right members *)
     match t with
-    | t when is_mutex_type t -> `Top
+    | t when is_mutex_type t -> `Bot (* use `Bot instead of `Top to avoid unknown value escape warnings *)
```

Comment:
That's a good point. I'm not sure about the details of the different struct domains, but in a couple of places there's some explicit workaround to this since arrays (and blobs) also initialize to `Bot`: https://github.com/goblint/analyzer/blob/45b2269ca56b1e6244f2c7bbb480cb49fb48a07c/src/cdomains/structDomain.ml#L159-L167

Maybe a safer alternative would be like a `Struct` with no fields? Of course introducing a new `Mutex` variant to the value domain would be the safest, but I suspect it requires similar hacks as for `Thread` since there's some unknown structure still being manipulated (and that structure probably is different for OSX).

The only reason I put this workaround here is to avoid large amounts of invariants about `pthread_mutex_t` contents, which are actually all incorrect, since base analysis never updates them from their initialized values. Hiding them from g2html has been an idea for a while as well, but so far it was only done for accesses, not base.
I can also revert this small change here and split the hiding of mutex contents into a separate PR for a more reliable and involved solution.

Reply:
> I can also revert this small change here and split the hiding of mutex contents into a separate PR for a more reliable and involved solution.

Yes, I think this might be the prudent thing to do!

Reply:
Reverted here and moved to #746.